_news_reader_project/app/src/main/java/com/manu/newsreader/ui/main/adapter/ArticleAdapter.java

```diff
@@ -0,0 +1,61 @@
+package com.manu.newsreader.ui.main.adapter;
+
+import android.annotation.SuppressLint;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.manu.newsreader.databinding.ArticleItemBinding;
+import com.manu.newsreader.ui.main.model.ArticleItemViewModel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ArticleAdapter extends RecyclerView.Adapter<ArticleAdapter.ArticleViewHolder> {
+
+    private List<ArticleItemViewModel> articleModelList;
+
+    public ArticleAdapter() {
+        this.articleModelList = new ArrayList<>();
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    public void setArticleModelList(List<ArticleItemViewModel> articleModelList) {
+        this.articleModelList = articleModelList;
+
+        // At first it receives an empty list and afterwards
+        // when the db query is done, it ignores the data received
```

Comment:
data implementation details are not relevant for UI layer (such as depicting the data is received from db query, repository interface is agnostic of the data source). Maybe we can add a better explanation to how this works